R/arrange_anova.R

```diff
@@ -60,21 +60,73 @@ arrange_anova.anova <- function(x) {
     x$term <- paste0("model", 2:nrow(object))
 
     class(x) <- c("apa_model_comp", class(x))
+  } else {
+    # --------------------------------------------------------------------------
+    # - stats::anova.lm()
+    # - car::leveneTest()
+    # - afex::mixed(method %in% c("KR", "S", "PB"))
+    # - lme4::summary.merMod()
+    # - lmerTest::summary.lmerModLmerTest()
+
+    # c("term", "sumsq", "df", "sumsq_err", "df_res", "statistic", "p.value")
+
+    # anova_table from mixed(method = "PB") contain two columns with *p* values,
+    # but also df from asymptotic theory
+    col_names <- colnames(object)
+    if("Chi Df" %in% col_names && "Pr(>PB)" %in% col_names) {
+      object$`Chi Df` <- NULL
+      object$`Pr(>Chisq)` <- NULL
+    }
+    if("Chi Df" %in% col_names && "Df" %in% col_names) {
+      object$Df <- NULL
+    }
 
-  } else if(is.null(object[["Sum Sq"]])) { # car::levenTest
-    x <- x[, -which(colnames(x) %in% c("sumsq", "sumsq_err"))]
+    renamers <- c(
```

Comment:
Do we maybe want to put this lookup table in a place of its own? I suspect this may be applicable more generally, no? Seems like a similar use case as [`convert_stat_name()`](https://github.com/crsh/papaja/blob/b94783a7c62643abfeeee57c6008622709378306/R/utils.R#L153).

Reply:
Love the idea! We could make the process more robust (e.g., by using `make.names()`) and transparent (one table for all, so it's like a dictionary). The only pro for separate lists would be that you can omit some column names that are not legal in a specific `apa_print()` method

However, I think a renaming function could take an argument with a whitelist of legal column names (that we anticipated while programming), maybe throwing a warning, not an error, when there are undetected columns.

Reply:
Yes, that was my thinking too. So we can have lookup methods for renaming columns, for labelling columns and so on. If there are any ambiguities we can still resolve those in place, I think.

Reply:
Not sure if I'm missing something but is this addressed somewhere?

Reply:
I implemented a new function `sanitize_table()` with lookup methods for both column names and labels on another branch which already works nicely with `htest` and `manova` objects; however, I thought it is wise to hold this back for another PR, because it might still be necessary to change the workflow a bit when we start using it throughout the package. Moreover, before deploying such strongly altered code, I wanted to add some more rigid testing.

Reply:
Alrighty, I hope you're keeping a list of all the other PRs you have in the pipeline. 😆